client/src/controllers/ActionController.test.js

```diff
@@ -22,7 +24,17 @@ describe('ActionController', () => {
       {
         ...Object.getOwnPropertyDescriptors(oldWindowLocation),
         assign: { configurable: true, value: jest.fn() },
-        reload: { configurable: true, value: jest.fn() },
+        reload: {
+          configurable: true,
+          value: jest.fn().mockImplementation(() => {
+            const event = new Event('beforeunload');
+            Object.defineProperty(event, 'returnValue', {
+              value: null,
+              writable: true,
+            });
```

Comment:
If we also handled the deprecated `returnValue` usage of the `beforeunload` event, then this mock can be simplified by checking whether `event.defaultPrevented` is `true` instead